hydromt/models/components/grid.py

```diff
@@ -683,99 +408,363 @@ def setup_grid(
             grid = grid.drop_vars("mask")
 
         # Add region and grid to model
-        self.set_geoms(geom, "region")
-        self.set_grid(grid)
+        # self.set_geoms(geom, "region") TODO: needs implemenation of ModelRegion
```

Comment:
Which is where you could need the weakref to model: self.model.region.set(geom)
We discussed with @savente93 that we will see how to keep region updated based on model main_component once we merge this PR and the ModelRegion PR together.